### DISCUSSION TITLE
Regarding SPDM over I2C

### DISCUSSION THREAD

**Discussion by H-Alshamsi** [DMTF/libspdm#1145] [Ideas]
We are working on SPDM application over I2C using two raspberry PI, we are wondering if there are work being done in similar manner.

**Comment by jyao1** [comment]
I dont know.

**Comment by Wenxing-hou** [comment]
Thanks for using libspdm to implement i2c project. 
But I don't know the SPDM application for i2c is based on [libspdm library code](https://github.com/DMTF/libspdm)  or [SPDM spec](https://www.dmtf.org/dsp/DSP0274) . 
Could you tell me?

**Reply by H-Alshamsi** [reply]
It's based on the libspdm emu code

**Comment by alokprasad7** [comment]
@H-Alshamsi You need to work on running RequesterEMU on one RPI and ResponderEMU on another , 
Need to hook function to send and rcv message to use i2c send /rcv calls. Read the code :)

**Comment by LeeTroy** [comment]
The latest kernel support socket-based MCTP and MCTP-over-SMBus binding.

- https://github.com/LeeTroy/meta-spdm/blob/main/recipes-example/spdm-emu/spdm-emu/0002-Enabling-MCTP-over-Kernel-Socket.patch
